cli/commands/download.js

```diff
@@ -0,0 +1,193 @@
+const fs = require("fs");
+const path = require("path");
+const axios = require("axios");
+
+var ProgressBar = require("progress");
+
+const { hasApiKeyForWorkspace, getApiKeyForWorkspace } = require("../core.js");
+
+const api = require("../../api.js");
+
+function parseDatasetUrlWithVersion(datasetUrl) {
+    if (datasetUrl.includes("/")) {
+        const parts = datasetUrl.split("/");
+
+        if (parts.length == 3) {
+            return {
+                workspaceUrl: parts[0],
+                projectUrl: parts[1],
+                version: parts[2]
+            };
+        } else if (parts.length == 2) {
+            return {
+                workspaceUrl: parts[0],
+                projectUrl: parts[1],
+                version: null
+            };
+        } else {
+            throw new Error("invalid project url");
+        }
+    } else {
+        return {
+            workspaceUrl: null,
+            datasetUrl: datasetUrl,
+            version: null
+        };
+    }
+}
+
+async function getProjectVersions(workspaceUrl, projectUrl, apiKey) {
+    const project = await api.getProject(workspaceUrl, projectUrl, apiKey);
+
+    return project.versions || [];
+}
+
+async function getLatestVersion(workspaceUrl, projectUrl, apiKey) {
+    const availableVersions = await getProjectVersions(workspaceUrl, projectUrl, apiKey);
+    availableVersions.sort((a, b) => a.created - b.created);
+
+    if (availableVersions.length > 0) {
+        const versionUrl = availableVersions[availableVersions.length - 1].id;
+        const parsed = parseDatasetUrlWithVersion(versionUrl);
+
+        return parsed.version;
+    }
+
+    return null;
+}
+
+async function verifyVersionExists(workspaceUrl, projectUrl, version, apiKey) {
+    const availableVersions = await getProjectVersions(workspaceUrl, projectUrl, apiKey);
+    const versionExists = availableVersions.find((v) => v.id.endsWith(`/${version}`));
+    if (!versionExists) {
+        throw new Error(`version ${version} does not exist for dataset ${projectUrl}`);
+    }
+}
+
+async function getDownloadLink(
+    workspaceUrl,
+    projectUrl,
+    version,
+    format,
+    apiKey,
+    recurseDepth = 0
+) {
+    if (recurseDepth > 12) {
```

Comment:
Let's set this to something higher. Would be cool if we can show a progress bar too.